crypto/dh/dh_ameth.c

```diff
@@ -215,7 +215,7 @@ static int dh_priv_encode(PKCS8_PRIV_KEY_INFO *p8, const EVP_PKEY *pkey)
     return 1;
 
  err:
-    OPENSSL_free(dp);
+    OPENSSL_clear_free(dp, dplen);
```

Comment:
Please move this to the error case of the `PKCS8_pkey_set0` call.

Reply:
> Please move this to the error case of the `PKCS8_pkey_set0` call.

Updated,Thanks

Reply:
> Thanks. You might also want to fix similar wrong code in dh and dsa.

Fixed,thanks